BUILD-8943 Create config-maven action (#93)
SonarSource/ci-github-actions: README.md

|--------------------|------------------------------------------------------------------------------------------------|
| `build-number`     | The current build number. Also set as environment variable `BUILD_NUMBER`                     |
| `snapshot-version` | The project version set in the pom.xml (before replacement). Also set as environment variable `CURRENT_VERSION` |
| `project-version`  | The project version with build number (after replacement). Also set as environment variable `PROJECT_VERSION` |

### Environment Variables Set

After running this action, the following environment variables are available:

- `BUILD_NUMBER`: The current build number
- `SNAPSHOT_VERSION`: The original project version from pom.xml
- `PROJECT_VERSION`: The project version with build number appended
- `ARTIFACTORY_USERNAME`: Username for Artifactory authentication
- `ARTIFACTORY_ACCESS_TOKEN`: Access token for Artifactory authentication
- `ARTIFACTORY_PASSWORD`: Deprecated alias for `ARTIFACTORY_ACCESS_TOKEN`
- `ARTIFACTORY_ACCESS_USERNAME`: Deprecated alias for `ARTIFACTORY_USERNAME`
- `MAVEN_OPTS`: JVM options for Maven execution

> - `SONARSOURCE_REPOSITORY_URL`: URL for SonarSource Artifactory repository

Reviewer: ```suggestion
- `SONARSOURCE_REPOSITORY_URL`: URL for SonarSource Artifactory root virtual repository (i.e.: sonarsource-qa for public builds or sonarsource-qa for private builds)
```

- `BASH_ENV`: Path to bash aliases file with Maven command alias

### Features

- **Automatic Build Number Management**: Integrates with `get-build-number` action for consistent build numbering
- **Project Version Configuration**: Automatically updates Maven project version with build number using format `<MAJOR>.<MINOR>.<PATCH>.<BUILD_NUMBER>`
- **Artifactory Integration**: Sets up authentication and repository configuration for SonarSource Artifactory
- **Maven Settings Management**: Provides pre-configured `settings.xml` with SonarSource repositories and authentication
- **Maven Local Repository Caching**: Automatically caches `~/.m2/repository` for faster builds
- **Common Maven Flags**: Sets up convenient Maven command alias with common flags for consistent execution

> - **Repository Visibility Detection**: Automatically configures appropriate Artifactory repositories based on repository visibility (public/private)

Reviewer: this duplicates https://github.com/SonarSource/ci-github-actions/pull/93/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R88-R94 no?

## `build-maven`

Build and deploy a Maven project with SonarQube analysis and Artifactory deployment.

### Requirements

> Required to run the [`config-maven`](#config-maven) action before running the `build-maven` action.

Reviewer: ```suggestion
Call to [`config-maven`](#config-maven) action is required before running the `build-maven` action.
```
or `It is required...`

Author: I'm removing this, because this is not a requirement, only a recommendation

#### Other Dependencies

- The Java and Maven tools must be pre-installed. Use of `mise` is recommended.

> - Required to use the `*.sonarsource.parent:parent` Maven parent.

Reviewer: ```suggestion
- The "Sonar parent POM" (`[org|com].sonarsource.parent:parent`) must be used. There's a public POM (org) and a private POM (com), respectively for public or private code.
```

| `public`                      | Whether to build and deploy with/to public repositories                                                                    | Auto-detected from repository visibility                             |
| `artifactory-reader-role`     | Suffix for the Artifactory reader role in Vault                                                                            | `private-reader` for private repos, `public-reader` for public repos |
| `artifactory-deployer-role`   | Suffix for the Artifactory deployer role in Vault                                                                          | `qa-deployer` for private repos, `public-deployer` for public repos  |
| `deploy-pull-request`         | Whether to deploy pull request artifacts                                                                                   | `false`                                                              |
| `maven-local-repository-path` | Path to the Maven cache directory, relative to the user home directory                                                     | `.m2/repository`                                                     |

> | `maven-opts`                  | Additional Maven options to pass to the build script (`MAVEN_OPTS`)                                                        | `-Xmx1536m -Xms128m`                                                 |

Reviewer: I'm really not sure about this. To me it does not obviously make sense to set the Maven OPTS in the config-maven action.
This is needed for the build, not the setup of the version. And it is very common to run multiple Maven commands with different OPTS, typically depending on the memory consumption by the tests.
